ykutil/src/addr.rs

```diff
@@ -81,11 +82,36 @@ pub fn off_to_vaddr(containing_obj: &Path, off: u64) -> Option<usize> {
     None // Not found.
 }
 
+// Given a virtual address in the current address space, (if possible) determine the name of the
+// symbol this belongs to, and the path to the object from which it came.
+//
+// On success returns `Ok((symbol_name, object_path))`, or on failure `Err(())`.
+//
+// This function uses `dladdr()` internally, and thus inherits the same symbol visibility rules
+// used there. For example, this function will not find unexported symbols.
+//
+// Note that the returned object path may not be canonical.
```

Comment:
What does this last sentence mean?

Reply:
As in, a [canonical path](https://doc.rust-lang.org/std/fs/fn.canonicalize.html).

We could clarify that this is just the absolute path.

Reply:
I know what a canonical path is, but it doesn't make sense here, as we may (are likely to be!) running on a different box where the paths no longer make sense other than (hopefully) unique IDs.

Reply:
I'm not sure i follow.

This function queries addresses from the same virtual address space as the currently running executable. Not sure how another machine factors into it. What am I missing?

Reply:
"Canonical paths" only make sense for actual files. We don't have the original files here (at least if I understand things correctly!): what we have is a binary which contains files. The paths we're getting here are the "full paths" of those files when they were compiled.

Reply:
Another detail, I added this comment because [one of my assertions](https://github.com/ykjit/yk/pull/560/files#diff-59272cdbf7ded73ed95ce99cc146e32b5e6ff445c7e18e8b12efed0fb0a90c1dR255-R261) failed when i tried to compare two equivalent, but non-canonical paths. The linker doesn't report canonical paths.

Reply:
Right, so we're using these paths as IDs, right? And they're not necessarily reliable IDs is what I think you're saying.

Reply:
Well, yes, but they are the only ID the linker (and LLVM IR) really speaks when it comes to identifying functions...

Reply:
Right, so I would suggest that we really need to be talking in terms of IDs, not canonical paths (but I'm willing to be proven wrong!). Does it mean that the comment should be "these IDs are not reliable" or ... ?

Reply:
I know what you are saying, but I dislike the term ID in this context.

Yes, a "canonical path" is a "unique ID", but a "canonical path" is a more accurate term here: all canonical paths are a unique ID, but not all unique IDs are a canonical path. Why use the more general term and leave room for misinterpretation?

Can we compromise on something like:

> Note that the returned object path may not uniquely identify an object file, since `dladdr(3)` doesn't necessarily return canonical paths.

Reply:
Or, maybe better, kill the sentence. We've already said this uses `dladdr(3)` under the hood, so a reader can know what to expect :)

Reply:
Killing the sentence works for me!